src/js/angular/ttyg/directives/chat-view.directive.js

```diff
@@ -0,0 +1,179 @@
+ *
+ * @example
+ * <chat-view></chat-view>
+ */
+function ChartViewComponent(toastr, $translate, TTYGContextService, TTYGService) {
+    return {
+        restrict: 'E',
+        templateUrl: 'js/angular/ttyg/templates/chat-view.html',
+        scope: {},
+        link: ($scope, element, attrs) => {
+
+            // =========================
+            // Public variables
+            // =========================
+
+            /**
+             * @type {ChatModel}
+             */
+            $scope.selectedChat = undefined;
+
+            /**
+             * @type {AgentModel}
+             */
+            $scope.selectedAgent = undefined;
+
+            /**
+             * @type {ChatQuestion}
+             */
+            $scope.chatQuestion = undefined;
+
+            // =========================
+            // Private variables
+            // =========================
+
+            // =========================
+            // Public functions
+            // =========================
+
+            /**
+             * Handles the ask question action.
+             */
+            $scope.ask = () => {
+                if (!$scope.chatQuestion.conversationId) {
+                    createNewChat();
+                } else {
+                    askQuestion();
+                }
+            };
+
+            // =========================
+            // Private functions
+            // =========================
+
+            const setupNewChatQuestion = () => {
+                $scope.chatQuestion = getEmptyChatQuestion();
+            };
+
+            const createNewChat = () => {
+                TTYGContextService.emit(TTYGEventName.CREATE_CHAT, $scope.chatQuestion);
+            };
+
+            const askQuestion = () => {
+                if ($scope.selectedChat) {
+                    $scope.selectedChat.messages.push(chatQuestionToChatMessageMapper($scope.chatQuestion));
+                }
+                const question = cloneDeep($scope.chatQuestion);
+                setupNewChatQuestion();
+                TTYGContextService.emit(TTYGEventName.ASK_QUESTION, question);
+            };
+
+            /**
+             * Handles answering a chat question.
+             *
+             * @param {ChatMessageModel} answer
+             */
+            const onQuestionAnswer = (answer) => {
+                if ($scope.selectedChat && $scope.selectedChat.id === answer.conversationId) {
+                    $scope.selectedChat.messages.push(answer);
+                }
+            };
+
+            /**
+             * Handles the change of the selected chat.
+             * @param {ChatModel} chat - the new selected chat.
+             */
+            const onSelectedChatChanged = (chat) => {
+                if (!chat) {
+                    $scope.selectedChat = undefined;
+                    $scope.chatQuestion = getEmptyChatQuestion();
```

Comment:
don't you have a method `setupNewChatQuestion` for this already?

Reply:
Done